src/main/java/gregtech/api/items/metaitem/stats/ItemFluidContainer.java

```diff
@@ -18,4 +50,97 @@ public ItemStack getContainerItem(ItemStack itemStack) {
         }
         return itemStack;
     }
+
+    @Override
+    public ActionResult<ItemStack> onItemUse(EntityPlayer player, World world, BlockPos pos, EnumHand hand,
+                                             EnumFacing facing, float hitX, float hitY, float hitZ) {
+        ItemStack stack = player.getHeldItem(hand);
+        if (!isBucket) return pass(stack);
+
+        var result = rayTrace(world, player);
+        if (result == null) return pass(stack);
+
+        ItemStack cellStack = GTUtility.copy(1, stack);
+        var cellHandler = FluidUtil.getFluidHandler(cellStack);
+        if (cellHandler == null) return pass(stack);
+
+        var cellFluid = cellHandler.drain(Integer.MAX_VALUE, false);
+        var blockHandler = FluidUtil.getFluidHandler(world, result.getBlockPos(), result.sideHit);
+        FluidStack soundFluid = cellFluid;
+        boolean success, isFill;
+
+        if (blockHandler == null) {
+            if (cellFluid == null || !cellFluid.getFluid().canBePlacedInWorld())
+                return pass(stack);
+
+            blockHandler = createHandler(cellFluid, world, pos.offset(facing));
+            success = GTTransferUtils.transferFluids(cellHandler, blockHandler) > 0;
+            isFill = true;
```

Comment:
Are these `isFill` backwards? Right now you are setting `isFill` to true when you are attempting to transfer fluid from the cell into the world, which is emptying, not filling.

Reply:
i interpreted it as "filling the block", but it really could be either or.